### PR TITLE
sci-electronics/minipro: version bump

### PR DESCRIPTION
**Pull request by salfter**
Version 0.5 has been out a while...the ebuild needed some changes to get it working.  I have it in my overlay for now, but it probably belongs here.

### CODE REVIEW

File: sci-electronics/minipro/minipro-0.5.ebuild [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 
-EAPI=7
+EAPI=8
 
 inherit bash-completion-r1 udev
 
 DESCRIPTION="A free and open TL866XX programmer"
 HOMEPAGE="https://gitlab.com/DavidGriffith/minipro"
 SRC_URI="https://gitlab.com/DavidGriffith/minipro/-/archive/${PV}/${P}.tar.gz"
+PATCHES="$FILESDIR/$P-makefile.patch"
```
Comment (AndrewAmmerlaan) [inline sci-electronics/minipro/minipro-0.5.ebuild:11]:
```suggestion
PATCHES=( "$FILESDIR/$P-makefile.patch" )
```

`PATCHES` should be an array.

File: sci-electronics/minipro/minipro-0.5.ebuild [hunk 2]
```diff
@@ -1,13 +1,14 @@
-# Copyright 1999-2020 Gentoo Authors
+# Copyright 1999-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=7
+EAPI=8
 
 inherit bash-completion-r1 udev
```
Comment (AndrewAmmerlaan) [inline sci-electronics/minipro/minipro-0.5.ebuild:6]: After these changes it looks like we no longer use `bash-completion-r1.eclass` so we can drop it from the inherit.

**Comment by AndrewAmmerlaan** [comment]
Looks good :+1: 

Could you please sign-off your commit as specified in Gentoo's [copyright policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin)?

**Comment by AndrewAmmerlaan** [comment]
I'm getting this compile failure when I try to build this version:

```
787:3: note: call to ‘free’ here
787 |   free(update_dat);
|   ^~~~~~~~~~~~~~~~
tl866iiplus.c:830:5: warning: pointer ‘update_dat’ may be used after ‘free’ [-Wuse-after-free]
830 |     free(update_dat);
|     ^~~~~~~~~~~~~~~~
tl866iiplus.c:787:3: note: call to ‘free’ here
787 |   free(update_dat);
|   ^~~~~~~~~~~~~~~~
```

**Comment by salfter** [comment]
> I'm getting this compile failure when I try to build this version:
> 
> ```
> >>> Compiling source in /var/tmp/portage/sci-electronics/minipro-0.5/work/minipro-0.5 ...
> make -j12 -l12 
> gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o xml.o xml.c
> gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o jedec.o jedec.c
> gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o ihex.o ihex.c
> gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o srec.o srec.c
> gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o database.o database.c
> Creating version.h
> gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o minipro.o minipro.c
> Creating version.c
> gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o tl866a.o tl866a.c
> gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o tl866iiplus.o tl866iiplus.c
> gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o usb_nix.o usb_nix.c
> gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o main.o main.c
> gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o version.o version.c
> main.c: In function ‘print_version_and_exit’:
> main.c:130:36: error: ‘GIT_DATE’ undeclared (first use in this function)
> 130 |   fprintf(stderr, output, VERSION, GIT_DATE, GIT_HASH, GIT_BRANCH);
> |                                    ^~~~~~~~
> main.c:130:36: note: each undeclared identifier is reported only once for each function it appears in
> make: *** [<builtin>: main.o] Error 1
> make: *** Waiting for unfinished jobs....
> tl866iiplus.c: In function ‘tl866iiplus_firmware_update’:
> tl866iiplus.c:824:5: warning: pointer ‘update_dat’ may be used after ‘free’ [-Wuse-after-free]
> 824 |     free(update_dat);
> |     ^~~~~~~~~~~~~~~~
> tl866iiplus.c:787:3: note: call to ‘free’ here
> 787 |   free(update_dat);
> |   ^~~~~~~~~~~~~~~~
> tl866iiplus.c:830:5: warning: pointer ‘update_dat’ may be used after ‘free’ [-Wuse-after-free]
> 830 |     free(update_dat);
> |     ^~~~~~~~~~~~~~~~
> tl866iiplus.c:787:3: note: call to ‘free’ here
> 787 |   free(update_dat);
> |   ^~~~~~~~~~~~~~~~```
> ```

I changed my CFLAGS from ```-pipe -O2 -march=native``` to ```-pipe -O0 -g -Wall``` to see if I could duplicate those results, but it still builds without error for me:

```
Calculating dependencies... done!

>>> Verifying ebuild manifests

>>> Emerging (1 of 1) sci-electronics/minipro-0.5::salfter
 * minipro-0.5.tar.gz SHA256 SHA512 WHIRLPOOL size ;-) ...               [ ok ]
>>> Unpacking source...
>>> Unpacking minipro-0.5.tar.gz to /var/tmp/portage/sci-electronics/minipro-0.5/work
>>> Source unpacked in /var/tmp/portage/sci-electronics/minipro-0.5/work
>>> Preparing source in /var/tmp/portage/sci-electronics/minipro-0.5/work/minipro-0.5 ...
 * Applying minipro-0.5-makefile.patch ...                               [ ok ]
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/sci-electronics/minipro-0.5/work/minipro-0.5 ...
>>> Source configured.
>>> Compiling source in /var/tmp/portage/sci-electronics/minipro-0.5/work/minipro-0.5 ...
make -j16 
gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o xml.o xml.c
gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o jedec.o jedec.c
gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o ihex.o ihex.c
gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o srec.o srec.c
gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o database.o database.c
Creating version.h
Creating version.c
gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o minipro.o minipro.c
gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o tl866a.o tl866a.c
gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o tl866iiplus.o tl866iiplus.c
gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o usb_nix.o usb_nix.c
gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o main.o main.c
gcc -g -O0 -Wall -DSHARE_INSTDIR="\"/usr/share/minipro\"" -I/usr/include/libusb-1.0    -c -o version.o version.c
gcc xml.o jedec.o ihex.o srec.o database.o minipro.o tl866a.o tl866iiplus.o version.o usb_nix.o main.o -lusb-1.0   -o minipro
>>> Source compiled.
>>> Test phase [not enabled]: sci-electronics/minipro-0.5

>>> Install sci-electronics/minipro-0.5 into /var/tmp/portage/sci-electronics/minipro-0.5/image
make -j16 DESTDIR=/var/tmp/portage/sci-electronics/minipro-0.5/image install 
mkdir -p /var/tmp/portage/sci-electronics/minipro-0.5/image/usr/bin
mkdir -p /var/tmp/portage/sci-electronics/minipro-0.5/image/usr/share/man/man1
mkdir -p /var/tmp/portage/sci-electronics/minipro-0.5/image/usr/share/minipro
cp minipro /var/tmp/portage/sci-electronics/minipro-0.5/image/usr/bin/
cp miniprohex /var/tmp/portage/sci-electronics/minipro-0.5/image/usr/bin/
cp infoic.xml /var/tmp/portage/sci-electronics/minipro-0.5/image/usr/share/minipro/
cp man/minipro.1 /var/tmp/portage/sci-electronics/minipro-0.5/image/usr/share/man/man1/
if [ -n "/lib/udev" ]; then \
        mkdir -p /var/tmp/portage/sci-electronics/minipro-0.5/image/lib/udev/rules.d; \
        cp udev/60-minipro.rules /var/tmp/portage/sci-electronics/minipro-0.5/image/lib/udev/rules.d/; \
        cp udev/61-minipro-plugdev.rules /var/tmp/portage/sci-electronics/minipro-0.5/image/lib/udev/rules.d/; \
        cp udev/61-minipro-uaccess.rules /var/tmp/portage/sci-electronics/minipro-0.5/image/lib/udev/rules.d/; \
fi
if [ -n "" ]; then \
        mkdir -p /var/tmp/portage/sci-electronics/minipro-0.5/image; \
        cp bash_completion.d/minipro /var/tmp/portage/sci-electronics/minipro-0.5/image/; \
fi
>>> Completed installing sci-electronics/minipro-0.5 into /var/tmp/portage/sci-electronics/minipro-0.5/image

 * Final size of build directory: 13284 KiB (12.9 MiB)
 * Final size of installed tree:  12408 KiB (12.1 MiB)

strip: x86_64-pc-linux-gnu-strip --strip-unneeded -N __gentoo_check_ldflags__ -R .comment -R .GCC.command.line -R .note.gnu.gold-version
   /usr/bin/minipro

>>> Installing (1 of 1) sci-electronics/minipro-0.5::salfter
 * Running udev control --reload for reloading rules and databases ...   [ ok ]
 * Running udev control --reload for reloading rules and databases ...   [ ok ]
>>> Auto-cleaning packages...

>>> No outdated packages were found on your system.

 * GNU info directory index is up-to-date.
```

I also deleted the cached source tarball...any change upstream should've invalidated the hash and kept it from building, but it just re-downloaded the same file and rebuilt it without issue.

File: sci-electronics/minipro/minipro-0.5.ebuild (hunk 1 above)
Comment (salfter) [reply sci-electronics/minipro/minipro-0.5.ebuild]: This change is now in the ebuild.

File: sci-electronics/minipro/minipro-0.5.ebuild (hunk 2 above)
Comment (salfter) [reply sci-electronics/minipro/minipro-0.5.ebuild]: This change is now in the ebuild.

**Comment by AndrewAmmerlaan** [comment]
> I changed my CFLAGS from `-pipe -O2 -march=native` to `-pipe -O0 -g -Wall` to see if I could duplicate those results, but it still builds without error for me:

Are you running a stable arch or an ~arch system? Which version of gcc are you using?

**Comment by salfter** [comment]
> Are you running a stable arch or an ~arch system?

Mostly stable, with selected packages enabled for ~amd64 in /etc/portage/package.accept_keywords.

> Which version of gcc are you using?

```gcc (Gentoo 11.3.0 p4) 11.3.0```

This is the stable version.

**Comment by thesamesam** [comment]
@AndrewAmmerlaan are you using a git hook for WORKDIR? sometimes I get failures like this because of mine

**Comment by AndrewAmmerlaan** [comment]
> @AndrewAmmerlaan are you using a git hook for WORKDIR? sometimes I get failures like this because of mine

Nope, or well at least I did not intentionally do git stuff in the WORKDIR.

I tried it again and now it works just fine, I'm not sure what was going on yesterday. Probably something wrong with my system that got resolved after this mornings upgrade.

**Comment by salfter** [comment]
...and now a bot has closed the PR without merging.  Do I need to start this all over again?

**Pull request closed without merging**